java/serving/src/test/java/feast/serving/it/ServingBase.java

```diff
@@ -84,14 +75,88 @@ static void initialize(DynamicPropertyRegistry registry) {
     environment.start();
   }
 
-  @BeforeAll
-  static void globalSetup() {
-    servingStub = TestUtils.getServingServiceStub(false, FEAST_SERVING_PORT, null);
+  @AfterAll
+  static void globalTeardown() {
+    environment.stop();
   }
 
-  @AfterAll
-  static void tearDown() throws Exception {
-    ((ManagedChannel) servingStub.getChannel()).shutdown().awaitTermination(10, TimeUnit.SECONDS);
+  @BeforeEach
```

Comment:
What's the reason to run this for each test?

Reply:
We need to create a new server instance with the new app configuration needed per test. Also makes the tests less flake-prone since it's a clean environment each time.